ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlSecurityContext.java

```diff
@@ -16,33 +16,32 @@
 package io.confluent.ksql.security;
 
 import io.confluent.ksql.services.ServiceContext;
-import java.security.Principal;
 import java.util.Optional;
 
 /**
  * A class that provides KSQL security related information for KSQL user requests.
  */
 public class KsqlSecurityContext {
-  private final Optional<Principal> userPrincipal;
+  private final Optional<KsqlPrincipal> userPrincipal;
   private final ServiceContext serviceContext;
 
   public KsqlSecurityContext(
-      final Optional<Principal> userPrincipal,
+      final Optional<KsqlPrincipal> userPrincipal,
       final ServiceContext serviceContext
   ) {
     this.userPrincipal = userPrincipal;
     this.serviceContext = serviceContext;
   }
 
   /**
-   * Returns a {@code java.security.Principal} object containing the name of the current
+   * Returns a {@code KsqlPrincipal} object containing the name and properties of the current
```

Comment:
nit: `{@code KsqlPrincipal}` should actually be `{@link KsqlPrincipal}`. Then, one can click on it to open the referred class.